tests/Unit/NumericalAlgorithms/Interpolation/Test_BarycentricRational.cpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include "NumericalAlgorithms/Interpolation/BarycentricRational.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <vector>
+
+namespace {
+template <class F>
+void test_barycentric_rational(const F& function, const double lower_bound,
+                               const double upper_bound, const size_t size,
+                               const size_t order) noexcept {
+  std::vector<double> x_values(size), y_values(size);
+  const double delta_x = (upper_bound - lower_bound) / size;
+  std::random_device rd;
+  const auto seed = rd();
+  CAPTURE(seed);
+  std::mt19937 gen(seed);
+  std::uniform_real_distribution<> dis(0.0, delta_x);
+  for (size_t i = 0; i < size; ++i) {
+    x_values[i] = lower_bound + i * delta_x + dis(gen);
+    y_values[i] = function(x_values[i]);
+  }
+
+  const auto lambda =
+      [ order, lower_bound, size, delta_x, &dis, &gen,
+        &function ](const intrp::BarycentricRational& interpolant) noexcept {
+    const auto deserialized_interpolant =
+        serialize_and_deserialize(interpolant);
+    Approx custom_approx = Approx::custom().epsilon(1.e-12).scale(1.0);
+    for (size_t i = 0; i < 10 * size; ++i) {
+      const double x_value = lower_bound + i * delta_x * 0.1 + 0.1 * dis(gen);
+      CAPTURE(x_value);
+      CHECK_ITERABLE_CUSTOM_APPROX(function(x_value), interpolant(x_value),
+                                   custom_approx);
+      CHECK_ITERABLE_CUSTOM_APPROX(
+          function(x_value), deserialized_interpolant(x_value), custom_approx);
+    }
+    CHECK(order == interpolant.order());
+    CHECK(order == deserialized_interpolant.order());
+  };
+
+  lambda(intrp::BarycentricRational{x_values, y_values, order});
```

Comment:
Ah! Boost's implementation has two constructors, one that takes iterators another that takes pointers. Do we want these? I like the idea of having the user of the class deal with getting from their data structure to a `std::vector`. If we have multiple constructors this can be called multiple times, otherwise I'll remove the lambda way of doing this. What are your thoughts?